### ISSUE TITLE
There is a cache issue after upgradation @nestjs version 9.2.1

### ISSUE THREAD

**Issue by shaileshvit** [nestjs/nest#11145]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

After upgradation @nestjs version 9.2.1, cache manager Get values are coming as undefined. Before upgradation it was working fine, however after upgradation in non-prod environment we are facing this issue. 

### Minimum reproduction code

await this.cacheManager.get("CacheKey");  

### Expected behavior

We have tried with both cache-manager version(V4 & V5) with below points.
If using cache-manager v4, provide ttl in seconds
If using cache-manager v5, provide ttl in milliseconds

Set the value and Get that value immediately working  fine, however after few seconds Get values are coming as undefined how ever ttl parameter is correct in millisecond.

import { CACHE_MANAGER, Inject, Injectable } from "@nestjs/common";
import { Cache } from "cache-manager";

await this.cacheManager.set("CacheKey", data, 1800000);

await this.cacheManager.get("CacheKey");  /// value coming as undefined after set value in few seconds. 

### Package

- [ ] I don't know. Or some 3rd-party package
- [X] <code>@nestjs/common</code>
- [X] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### Other package

npm  cache-manager: 4.1.0  or  cache-manager: 5.1.6

### NestJS version

9.2.1

### Packages versions

"dependencies": {
    "@azure/identity": "^3.1.2",
    "@azure/keyvault-keys": "^4.6.0",
    "@azure/keyvault-secrets": "^4.6.0",
    "@nestjs/axios": "1.0.1",
    "@nestjs/common": "^9.2.1",
    "@nestjs/config": "^2.2.0",
    "@nestjs/core": "^9.2.1",
    "@nestjs/platform-express": "^9.2.1",
    "@nestjs/swagger": "^6.1.4",
    "@nestjs/typeorm": "^9.0.1",
    "cache-manager": "^5.1.6",
    "class-transformer": "^0.5.1",
    "class-validator": "^0.14.0",
    "contentful": "^9.1.5",
    "contentful-management": "^10.6.3",
    "copy-files-from-to": "^3.6.0",
    "helmet": "^6.0.1",
    "hpp": "^0.2.3",
    "lodash": "^4.17.21",
    "marked": "^4.0.12",
    "moment": "^2.29.4",
    "moment-timezone": "^0.5.40",
    "mssql": "^9.0.1",
    "nest-winston": "^1.8.0",
    "query-string": "^7.1.1",
    "redact-pii-fork": "^1.0.0",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^3.0.2",
    "rxjs": "^7.8.0",
    "typeorm": "^0.3.11",
    "uuid": "^8.3.2",
    "winston": "^3.8.2"
  },
  "devDependencies": {
    "@nestjs/cli": "^9.1.8",
    "@nestjs/schematics": "^9.0.4",
    "@nestjs/testing": "^9.2.1",
    "@types/express": "^4.17.15",
    "@types/jest": "29.2.5",
    "@types/node": "^18.11.18",
    "@types/supertest": "^2.0.12",
    "@typescript-eslint/eslint-plugin": "^5.48.0",
    "@typescript-eslint/parser": "^5.48.0",
    "eslint": "^8.31.0",
    "eslint-config-prettier": "^8.6.0",
    "eslint-plugin-prettier": "^4.2.1",
    "jest": "29.3.1",
    "prettier": "^2.8.2",
    "supertest": "^6.3.3",
    "swagger-ui-express": "^4.1.2",
    "ts-jest": "29.0.3",
    "ts-loader": "^9.4.2",
    "ts-node": "^10.9.1",
    "tsconfig-paths": "^4.1.2",
    "typescript": "^4.9.4"
  }

### Node.js version

16.13.2

### In which operating systems have you tested?

- [ ] macOS
- [ ] Windows
- [X] Linux

### Other

Locally and in production environment it is working fine. In non-prod it is not working. We are using docker based deployment for non-prod and prod using Linux OS.

**Comment by micalevisk** [comment]
Hi! Thanks for reporting

As you said that it works depending on your environment, how can we ensure that this is not some misconfiguration on your side (especially docker stuff)? Have you tried to remove the lock file and installing everything again? Also, from which version have you upgraded? 

questions that could be easily answered with a minimum reproduction repository...

**Comment by shaileshvit** [comment]
> 

Hello Micalevisk,
Thanks for looking into this.
Previous version was @nestjs: 8.0.0 and upgraded to @nestjs: 9.2.1
I have tried with your suggestion to remove the lock file and generated it again. 
We have not did any major misconfiguration during upgradation instead of docker file update. Docker file basically we have updated for installing the native packages in the container. Sharing here docker file and other codes.
Docker file ...

FROM node:18.1.0-buster-slim

# Create app directory
WORKDIR /usr/src/app
ARG NPM_TOKEN
# Install app dependencies
# A wildcard is used to ensure both package.json AND package-lock.json are copied
# where available (npm@5+)
COPY package*.json ./
COPY .npmrc .npmrc
RUN apt-get update -y
# RUN apt-get install libsecret-1-dev -y
# If you are building your code for production 

RUN apt-get update && \
    apt-get install -y \
    apt-utils chromium
RUN npm install
# Bundle app source
COPY . .

EXPOSE 3000
ENV PORT 3000

RUN npm run build 
# RUN npm run build

# Health check
HEALTHCHECK --interval=5m --timeout=5s CMD curl--fail http://localhost:3000 || exit 1

CMD [ "node", "dist/main" ]

======================================

Package.json

{
  "name": "dc-us-en-api",
  "version": "0.0.1",
  "description": "",
  "author": "",
  "private": true,
  "license": "UNLICENSED",
  "scripts": {
    "prebuild": "rimraf dist",
    "build": "nest build",
    "postbuild": "copy-files-from-to --config fileCopyConfig.json",
    "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
    "start": "copy-files-from-to --config fileCopyConfig.json && nest start",
    "start:dev": "copy-files-from-to --config fileCopyConfig.json && nest start --watch",
    "start:debug": "nest start --debug --watch",
    "start:prod": "copy-files-from-to --config fileCopyConfig.json && node dist/main",
    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\"",
    "lint:fix": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
    "eslint-report": "eslint \"{src,apps,libs,test}/**/*.ts\" -o report.json -f json",
    "check-quality": "npm run lint",
    "check-quality:fix": "npm run lint:fix",
    "test": "jest --passWithNoTests",
    "test:watch": "jest --watch",
    "test:coverage": "jest --coverage",
    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
    "test:e2e": "jest --config ./test/jest-e2e.json"
  },
  "dependencies": {
    "@azure/identity": "^3.1.2",
    "@azure/keyvault-keys": "^4.6.0",
    "@azure/keyvault-secrets": "^4.6.0",
    "@dentalcare/generate-html-pdf": "^2.0.8",
    "@dentalcare/modify-pdf": "^1.0.5",
    "@nestjs/axios": "1.0.1",
    "@nestjs/common": "^9.2.1",
    "@nestjs/config": "^2.2.0",
    "@nestjs/core": "^9.2.1",
    "@nestjs/platform-express": "^9.2.1",
    "@nestjs/swagger": "^6.1.4",
    "@nestjs/typeorm": "^9.0.1",
    "cache-manager": "^5.1.6",
    "class-transformer": "^0.5.1",
    "class-validator": "^0.14.0",
    "contentful": "^9.1.5",
    "contentful-management": "^10.6.3",
    "copy-files-from-to": "^3.6.0",
    "helmet": "^6.0.1",
    "hpp": "^0.2.3",
    "lodash": "^4.17.21",
    "marked": "^4.0.12",
    "moment": "^2.29.4",
    "moment-timezone": "^0.5.40",
    "mssql": "^9.0.1",
    "nest-winston": "^1.8.0",
    "query-string": "^7.1.1",
    "redact-pii-fork": "^1.0.0",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^3.0.2",
    "rxjs": "^7.8.0",
    "typeorm": "^0.3.11",
    "uuid": "^8.3.2",
    "winston": "^3.8.2"
  },
  "devDependencies": {
    "@nestjs/cli": "^9.1.8",
    "@nestjs/schematics": "^9.0.4",
    "@nestjs/testing": "^9.2.1",
    "@types/express": "^4.17.15",
    "@types/jest": "29.2.5",
    "@types/node": "^18.11.18",
    "@types/supertest": "^2.0.12",
    "@typescript-eslint/eslint-plugin": "^5.48.0",
    "@typescript-eslint/parser": "^5.48.0",
    "eslint": "^8.31.0",
    "eslint-config-prettier": "^8.6.0",
    "eslint-plugin-prettier": "^4.2.1",
    "jest": "29.3.1",
    "prettier": "^2.8.2",
    "supertest": "^6.3.3",
    "swagger-ui-express": "^4.1.2",
    "ts-jest": "29.0.3",
    "ts-loader": "^9.4.2",
    "ts-node": "^10.9.1",
    "tsconfig-paths": "^4.1.2",
    "typescript": "^4.9.4"
  }
}
==========================================

app.module.ts

import {
  CacheInterceptor,
  MiddlewareConsumer,
  Module,
  NestModule,
  CacheModule,
} from "@nestjs/common";
import { APP_FILTER, APP_INTERCEPTOR } from "@nestjs/core";
import { ConfigService } from "@nestjs/config";
import { AppController } from "src/app.controller";
import { AppService } from "src/app.service";
import { LoggerMiddleware } from "src/middleware/logger.middleware";

@Module({
  imports: [
    CacheModule.register({isGlobal: true}),
  ],

  controllers: [AppController],
  providers: [
    AppService,   
    {
      provide: APP_INTERCEPTOR,
      useClass: CacheInterceptor,
    },
  ],
})
export class AppModule implements NestModule {
  /*constructor(private configService: ConfigService) {}
  configure(consumer: MiddlewareConsumer) {
    consumer.apply(LoggerMiddleware).forRoutes("*");
  }*/
}
===================================================
app.controller.ts

import { Controller, Get, UseInterceptors, CacheInterceptor, CacheTTL } from "@nestjs/common";
import { ApiOperation, ApiTags } from "@nestjs/swagger";
import { AppService } from "./app.service";
import { TTL } from "./enum/cache.enum";

@ApiTags("Dentalcare")
@Controller()
@UseInterceptors(CacheInterceptor)
export class AppController {
  constructor(private readonly appService: AppService) {}
  @Get()
  @CacheTTL(TTL.ONE_MINUTE)
  @ApiOperation({
    summary: "Base API",
  })
  getHello(): string {
    return this.appService.getHello();
  }
}
===================================================
export enum TTL {
    TEN_SECONDS = 1000 * 10,
    ONE_MINUTE = 1000 * 60,
    HALF_HOUR = 1000 * 60 * 30,
    ONE_HOUR = 1000 * 60 * 60,
    TWO_HOUR = 1000 * 2 * 60 * 60,
    THREE_HOUR = 1000 * 3 * 60 * 60,
    ONE_DAY = 1000 * 60 * 60 * 24,
    ONE_WEEK = 1000 * 7 * 24 * 60 * 60,
  }
===================================================

files where we have two api calls from controller and service methods for set and get cache data.
course.controller.ts

// In this API call we set the cache value.
 @Post("evaluate-test-result")
  @ApiOperation({ summary: "Evaluate course test result." })
  @ApiBody({ type: EvaluationDTO })
  @ApiCreatedResponse()
  async createEvaluation(@Body() evaluationDTO: EvaluationDTO) {
    return await this.courseService.evaluateTestResult(evaluationDTO);
  }

// In this API call we trying to read cache value where it is getting failed sometimes and coming as undefined.
@ApiOperation({ summary: "Insert course-test result work flow." })
  @Post("save-course-test-result")
  @HttpCode(200)
  @ApiBody({ type: CourseTestStepsDTO })
  @ApiCreatedResponse({ type: CourseTestStepsDTO })
  async saveCourseTesResultWorkFlow(
    @Body() courseTestStepsDto: CourseTestStepsDTO,
  ) {
    return await this.courseService.saveCourseTesResultWorkFlow(
      courseTestStepsDto,
    );
  }

course.service.ts

async evaluateTestResult(evaluationDTO: EvaluationDTO) {
      const ttl = TTL.HALF_HOUR;
      await this.cacheManager.set("sample_key", sessionData, ttl);
}

async saveCourseTesResultWorkFlow(courseTestStepsDto: CourseTestStepsDTO) {
     const sessionData: any = await this.cacheManager.get("sample_key"); // Value sometime coming as undefined.
}

**Comment by micalevisk** [comment]
@shaileshvit could you please edit your message? it's quite hard to follow. Read [this](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) if you don't know how to create fenced code blocks.

**Comment by shaileshvit** [comment]
Hello Micalevisk<https://github.com/micalevisk>,

Sorry, now edited, please check.

Thanks And Regards,
Shailesh Kumar
TECHNICAL LEAD
Upcoming OOO :
Ph: +91 9620261495
***@***.*** ***@***.***

**Comment by kamilmysliwiec** [comment]
Please, use our [Discord](https://discord.gg/G7Qnnhy) channel (support) for such questions. We are using GitHub to track bugs, feature requests, and potential improvements.
